src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -524,25 +526,35 @@ private bool TryGetUnmanagedCallingConventionFromModOpt(MethodSignature signatur
                 if (defType.Namespace != "System.Runtime.CompilerServices")
                     continue;
 
-                // Take the first recognized calling convention in metadata.
+                void SetCallConv(CorInfoCallConv toSet)
```

Comment:
This is very clever. I am personally not a fan, but defer to the crossgen2 owners.

Reply:
The convention for functions like this is to put them after the return statement of the function wherever possible. When you embed them in the middle of the function its easy to think they are part of the natural program flow.

Also, its very unclear for a reader where callConvLocal gets set. I would make that a ref or out parameter to the helper function, if only to make it clear about how the data flows through the function.